lib/utils.js

```diff
@@ -257,12 +259,13 @@ async function getXctestrunFilePath (deviceInfo, sdkVersion, bootstrapPath) {
  * Return the name of xctestrun file
  * @param {DeviceInfo} deviceInfo
  * @param {string} version - The Xcode SDK version of OS.
+ * @param {string} architecture - The architecture of device.
  * @return {string} returns xctestrunFilePath for given device
  */
-function getXctestrunFileName (deviceInfo, version) {
+function getXctestrunFileName (deviceInfo, version, architecture = 'x86_64') {
   return isTvOS(deviceInfo.platformName)
-    ? `WebDriverAgentRunner_tvOS_appletv${deviceInfo.isRealDevice ? `os${version}-arm64` : `simulator${version}-x86_64`}.xctestrun`
```

Comment:
Perhaps you could use:
```
`simulator${version}-${os.arch() === "arm64" ? "arm64" : "x86_64"}`
```
instead of `x86_64`?
The simulator depends on the host, thus this line can guess the proper naming with the host's architecture.

Reply:
I agree. The WDA architecture should be selected based on the current host architecture

Reply:
I see we can check the arch by `os.arch()`, thank you for letting me know about it. I'll fix it.

Reply:
By the way, where can I get the `os` variable? And if it contains the string directly, we may use it like
```
`simulator${version}-${os.arch()"}`
```

I checked the `DeviceInfo` but it doesn't contain the property.

And I also noticed I should change the comment here based on the fix.
https://github.com/appium/WebDriverAgent/blob/master/lib/utils.js#L160-L184

Reply:
It is https://nodejs.org/api/os.html